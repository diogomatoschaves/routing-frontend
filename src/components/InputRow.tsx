import React, { Component, useState } from 'react'
import { Box } from '../styledComponents'
import { Coords, UpdatePoint } from '../types'
import { NORMAL_INPUT } from '../utils/colours'
import BackgroundIcon from './BackgroundIcon'
import ControlledInput from './ControlledInput'

interface Props {
  rowKey: string
  index: number
  coords: Coords
  placeholder: string
  iconName: string | any
  updatePoint: UpdatePoint
  urlMatchString: string
}

const diameter = 38

const InputRow = (props: Props) => {
  const [color, setColor] = useState(NORMAL_INPUT)

  const {
    iconName,
    rowKey,
    index,
    coords,
    updatePoint,
    placeholder,
    urlMatchString
  } = props

  return (
    <Box direction="row" justify="space-around" padding="10px 0">
      <BackgroundIcon
        diameter={diameter}
        color={color}
        iconColor={'white'}
        circle={true}
        iconName={iconName}
        margin={'0 10px 0 0'}
      />
      <ControlledInput
        rowKey={rowKey}
        index={index}
        coords={coords}
        updatePoint={updatePoint}
        updateColor={setColor}
        placeholder={placeholder}
        urlMatchString={urlMatchString}
        color={color}
      />
    </Box>
  )
}

export default InputRow
