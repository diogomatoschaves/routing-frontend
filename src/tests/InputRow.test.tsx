import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import TestRenderer from 'react-test-renderer'
import InputRow from '../components/InputRow'

const urlMatchString = '/:profile/:start/:end'

it('expect to render Panel component', () => {
  const item = {
    lat: null,
    lon: null,
    marker: 'map marker alternate',
    name: 'start',
    placeholder: 'Origin'
  }

  const testInstance = TestRenderer.create(
    <MemoryRouter initialEntries={['/']}>
      <InputRow
        rowKey={item.name}
        index={0}
        coords={{ lat: 52, lon: 12 }}
        placeholder={item.placeholder}
        iconName={item.marker}
        updatePoint={jest.fn()}
        urlMatchString={urlMatchString}
      />
    </MemoryRouter>
  )
  expect(testInstance.toJSON()).toMatchSnapshot()
})
