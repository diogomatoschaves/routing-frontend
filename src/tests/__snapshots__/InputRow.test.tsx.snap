// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Panel component 1`] = `
<div
  className="sc-bdVaJa gSlpgJ"
  direction="row"
>
  <div
    className="sc-gZMcBi fWaAxO"
  >
    <div
      className="sc-bwzfXH fNQCHt"
      color="#007a93"
    >
      <div
        className="sc-bdVaJa cCYAHW"
        height="100%"
      >
        <i
          aria-hidden="true"
          className="map marker alternate icon sc-htpNat fQDcBg"
          fontSize="25px"
          height="20px"
          onClick={[Function]}
          overridecolor="white"
          position="absolute"
        />
      </div>
    </div>
  </div>
  <div
    className="ui fluid icon input sc-dnqmqq jwbNvq"
  >
    <input
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Origin"
      type="text"
      value="52, 12"
    />
    <i
      aria-hidden="true"
      className="remove circle link icon"
      onClick={[Function]}
    />
  </div>
</div>
`;
